specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,267 @@
+    }
+    else
+    {
+        // Register a handler for the `SaveAsRequested` event.
+        m_webView2Staging20->add_SaveAsRequested(
+            Callback<ICoreWebView2StagingSaveAsRequestedEventHandler>(
+                [this](
+                    ICoreWebView2* sender,
+                    ICoreWebView2StagingSaveAsRequestedEventArgs* args) -> HRESULT
+                {
+                    // Hide the system default save as dialog.
+                    CHECK_FAILURE(args->put_Handled(TRUE));
+
+                    auto showCustomizedDialog = [this, args]
+                    {
+                        // Preview the content mime type, optional
+                        wil::unique_cotaskmem_string mimeType;
+                        CHECK_FAILURE(args->get_ContentMimeType(&mimeType));
+
+                        SaveAsDialog dialog(m_appWindow->GetMainWindow(), contentSaveTypes);
+                        if (dialog.confirmed)
+                        {
+                            // Set the ResultFilePath, SaveAsType, AllowReplace for the event
+                            // args from this customized dialog inputs, optional. If nothing
+                            // needs to input, the event args will provide default values.
+                            CHECK_FAILURE(
+                                args->put_ResultFilePath((LPCWSTR)dialog.path.c_str()));
+                            CHECK_FAILURE(args->put_SaveAsType(dialog.selectedType));
+                            CHECK_FAILURE(args->put_AllowReplace(dialog.allowReplace));
+
+                            // Confirm to download, required
+                            CHECK_FAILURE(args->put_ConfirmToSave(TRUE));
+                        }
+                    };
+
+                    wil::com_ptr<ICoreWebView2Deferral> deferral;
+                    CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                    m_appWindow->RunAsync(
+                        [deferral, showCustomizedDialog]()
+                        {
+                            showCustomizedDialog();
+                            CHECK_FAILURE(deferral->Complete());
+                        });
+                    return S_OK;
+                })
+                .Get(),
+            &m_saveAsRequestedToken);
+    }
+    m_silentSaveAs = !m_silentSaveAs;
+    MessageBox(
+        m_appWindow->GetMainWindow(),
+        (m_silentSaveAs ? L"Silent Save As Enabled" : L"Silent Save As Disabled"), L"Info",
+        MB_OK);
+    return true;
+}
+//! [ToggleSilent]
+
+//! [ProgrammaticSaveAs]
+// Call SaveContentAs method to trigger the programmatic save as.
+bool ScenarioSaveAs::ProgrammaticSaveAs()
+{
+    if (!m_webView2Staging20)
+        return false;
+    m_webView2Staging20->SaveContentAs(
+        Callback<ICoreWebView2StagingSaveContentAsCompletedHandler>(
+            [this](HRESULT errorCode, COREWEBVIEW2_SAVE_CONTENT_AS_RESULTS result) -> HRESULT
```

Comment:
What do apps want to do after saving the document? Do they not need to know where the document was saved on disk?

Reply:
> If they use this method to silent save as, they know and assign it from the event args

That's true, although a lot of effort to get the path and you have to get it from the event handler not where you called SaveContentAs.

> If they use this method to just open the system dialog, they will know and assign the location there.

That doesn't make sense to me. How do you get the location if they use the system dialog?

But also, my question was less if its possible to get the location, but if we think its common for end devs to want to do Save As and then know where the file was saved. If so, it would be much easier to get the location out of the async result of SaveContentAs.

Reply:
Can we add a way for the Save As API to be used with the Downloads API? Can we give them the started DownloadOperation as a result of the SaveContentAs method call so they know when the DownloadStarted event is raised that its for their SaveContentAs call?